stdlib/collections/__init__.pyi

```diff
@@ -37,7 +37,19 @@ else:
 
 class UserDict(MutableMapping[_KT, _VT]):
     data: dict[_KT, _VT]
-    def __init__(self, __dict: Mapping[_KT, _VT] | None = ..., **kwargs: _VT) -> None: ...
+    # __init__ should be kept roughly in line with `dict.__init__`, which has the same semantics
+    @overload
+    def __init__(self: UserDict[_KT, _VT], __dict: None = ...) -> None: ...
+    @overload
+    def __init__(self: UserDict[str, _VT], __dict: None = ..., **kwargs: _VT) -> None: ...
+    @overload
+    def __init__(self, __dict: SupportsKeysAndGetItem[_KT, _VT], **kwargs: _VT) -> None: ...
+    @overload
+    def __init__(self, __iterable: Iterable[tuple[_KT, _VT]], **kwargs: _VT) -> None: ...
+    # Next overload is for UserDict(string.split(sep) for string in iterable)
+    # Cannot be Iterable[Sequence[_T]] or otherwise UserDict(["foo", "bar", "baz"]) is not an error
```

Comment:
I don't think we need this comment, as the overload is copied from `builtins.pyi` and already has a comment there.